dynamiqs/time_qarray.py

```diff
@@ -679,52 +723,49 @@ def layout(self) -> Layout:
     @property
     def mT(self) -> TimeQArray:
         f = jtu.Partial(lambda t: self.f(t).mT)
-        return CallableTimeQArray(f, self._disc_ts)
+        return self._replace(f=f)
 
     @property
     def in_axes(self) -> PyTree[int | None]:
-        return CallableTimeQArray(0, None)
+        return CallableTimeQArray(None, None, 0, None)
 
     @property
     def discontinuity_ts(self) -> Array | None:
         return self._disc_ts
 
     def reshape(self, *shape: int) -> TimeQArray:
         f = self.f.reshape(*shape)
-        return CallableTimeQArray(f, self._disc_ts)
+        return self._replace(f=f)
 
     def broadcast_to(self, *shape: int) -> TimeQArray:
         f = self.f.broadcast_to(*shape)
-        return CallableTimeQArray(f, self._disc_ts)
+        return self._replace(f=f)
 
     def conj(self) -> TimeQArray:
         f = self.f.conj()
-        return CallableTimeQArray(f, self._disc_ts)
+        return self._replace(f=f)
 
-    def __call__(self, t: ScalarLike) -> QArray:
+    def _call(self, t: ScalarLike) -> QArray:
         return self.f(t)
 
     def __mul__(self, y: QArrayLike) -> TimeQArray:
         f = self.f * y
-        return CallableTimeQArray(f, self._disc_ts)
-
-    def __add__(self, y: QArrayLike | TimeQArray) -> TimeQArray:
-        if isinstance(y, get_args(ScalarLike)):
-            return ConstantTimeQArray(self.f + y)
-        elif isqarraylike(y):
-            y = ConstantTimeQArray(asqarray(y))
-            return SummedTimeQArray([self, y])
-        elif isinstance(y, TimeQArray):
-            return SummedTimeQArray([self, y])
-        else:
-            return NotImplemented
+        return self._replace(f=f)
 
 
 class SummedTimeQArray(TimeQArray):
     timeqarrays: list[TimeQArray]
 
-    def __init__(self, timeqarrays: list[TimeQArray], check: bool = True):
-        if check:
+    def _replace(self, timeqarrays: list[TimeQArray] | None = None) -> TimeQArray:
+        if timeqarrays is None:
+            timeqarrays = self.timeqarrays
+        return super()._replace(timeqarrays=timeqarrays)
+
+    def __init__(self, tstart: float, tend: float, timeqarrays: list[TimeQArray]):
+        super().__init__(tstart, tend)
+
+        # dirty way to bypass the check for the implementation of `in_axes`
+        if tstart is not None and tend is not None:
```

Comment:
Can't we put this check/shape broadcast inside a `__post_init__` (or maybe `__check_init__`) for this? I don't quite understand the problem of `in_axes` I think.